### PR TITLE
AP_Soaring: Independently specify thermalling bank

### PR DESCRIPTION
**Pull request by samuelctabor**
This adds a parameter SOAR_THML_BANK to specify the average bank angle to use when thermalling. This has two benefits - 

 - makes thermal turns independent from WP_LOITER_RAD, so thermal turns can be tight while having a large, safe WP_LOITER_RAD for general operations.
 - specifying as a bank angle makes the default more likely to be suitable over a range of vehicle speeds, so reduces user setup.

### CODE REVIEW

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (Hwurzburg) [inline libraries/AP_Soaring/AP_Soaring.cpp:137]: seems to be pretty narrow range for bank...is it really this restricted?

Comment (samuelctabor) [reply libraries/AP_Soaring/AP_Soaring.cpp:137]: This is the recommended range, as always you can set outside this range by accepting the warning. Note also that this is the average bank, so less or more bank up to LIM_ROLL_CD can be used for centering corrections

**Comment by samuelctabor** [comment]
https://github.com/ArduPilot/ardupilot_wiki/pull/3013 updated

File: ArduPlane/mode_thermal.cpp
Comment (magicrub) [inline ArduPlane/mode_thermal.cpp:118]: const float

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (magicrub) [reply libraries/AP_Soaring/AP_Soaring.cpp:137]: "Roll" is used more in this library than "bank" so lets stick with that terminology

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (magicrub) [inline libraries/AP_Soaring/AP_Soaring.cpp:486]: don't need to pull _ahrs out as a variable, just use it inline

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (magicrub) [inline libraries/AP_Soaring/AP_Soaring.cpp:488]: const float

Also, please make the squaring more obvious with (target_aspd*target_aspd)

**Review by magicrub — COMMENTED** [review]
minor tweaks needed but otherwise LGTM functionally

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (magicrub) [reply libraries/AP_Soaring/AP_Soaring.cpp:137]: although... there's NAVL1_LIM_BANK, which this is overriding so lets name this SOAR_LIM_BANK and in the description mention that it overrides NAVL1_LIM_BANK and if this is new param is set to 0 then it uses NAVL1_LIM_BANK.

Comment (samuelctabor) [reply libraries/AP_Soaring/AP_Soaring.cpp]: - Widened the limits to 20 - 50deg.
- Kept the naming as-is since it's not a limit but rather a target. Removed misleading left-over comment in mode_thermal that I expect was causing confusion

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (magicrub) [inline libraries/AP_Soaring/AP_Soaring.cpp:138]: add
```suggestion
    // @Units: deg 
```

Comment (magicrub) [inline libraries/AP_Soaring/AP_Soaring.cpp:138]:
```suggestion
    // @User: Advanced
    // @Units: deg
```

**Review by magicrub — APPROVED** [review]

**Comment by magicrub** [comment]
I added units to the param, ready to merge after it passes CI

**Pull request merged**
